libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx

```diff
@@ -148,50 +145,34 @@ export function CrudModalFeature({ application, onClose, port, organizationId, p
     },
     mode: 'onChange',
   })
-  const dispatch = useDispatch<AppDispatch>()
-
-  const toasterCallback = () => {
-    if (application) {
-      dispatch(
-        postApplicationActionsRedeploy({
-          applicationId: application?.id || '',
-          environmentId: application?.environment?.id || '',
-          serviceType: getServiceType(application),
-          callback: () =>
-            navigate(
-              ENVIRONMENT_LOGS_URL(organizationId, projectId, application?.environment?.id) +
-                DEPLOYMENT_LOGS_URL(application?.id)
-            ),
-          queryClient,
-        })
-      )
-    }
-  }
 
-  const onSubmit = methods.handleSubmit((data) => {
-    if (!application) return
+  const onSubmit = methods.handleSubmit(async (data) => {
+    if (!service) return
 
-    setLoading(true)
-    const cloneApplication = handleSubmit(data, application, port)
+    const cloneApplication = handleSubmit(data, service, port)
```

Comment:
I think you can use generics to properly type request and remove `as` cast.
`handleSubmit` signature should look something like that:
```ts
function handleSubmit<T extends Application | Container, R = T extends Application ? ApplicationEditRequest : ContainerRequest>({
    internal_port,
    external_port,
    publicly_accessible,
    protocol,
    name,
  }: {
    internal_port: string | undefined
    external_port: string | undefined
    publicly_accessible: boolean
    protocol: PortProtocolEnum | undefined
    name: string | undefined
  }, application: T, currentPort?: ServicePort): R {
```

Reply:
Yes for simple operation I know I can do that but here we need to update probes for Healthchecks, and I don't want to update all the logic in this PR just only React Query that why I keep the `handleSubmit`

Reply:
https://github.com/Qovery/console/pull/1085/commits/81738b0fd981cd8b229870fa3df15f3816998a8f